arch/risc-v/src/qemu-rv/qemu_rv_start.c

```diff
@@ -49,6 +49,34 @@
 #  error "Target requires kernel in S-mode, enable CONFIG_ARCH_USE_S_MODE"
 #endif
 
+/****************************************************************************
+ * Extern Function Declarations
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+extern void __trap_vec(void);
+extern void __trap_vec_m(void);
+extern void up_mtimer_initialize(void);
```

Comment:
should we make it as a general infrastructure?

Reply:
@xiaoxiang781216 
I'm still not sure if this approach is suitable for other RISC-V SoCs.

Reply:
Ok, let's do it later.